src/timeout.py

```diff
@@ -22,26 +22,27 @@
 
 class TimeoutService(Service):
 
-    PENDING_STATES = ['running', 'available', 'closing']
-
     def __init__(self, configs, args, name):
         super().__init__(configs, args, name)
         self._poll_period = args.poll_period
+        self._node_states = self._db.node_states
 
     def _get_pending_nodes(self, filters=None):
         nodes = {}
         node_filters = filters.copy() if filters else {}
-        for state in self.PENDING_STATES:
-            node_filters['state'] = state
+        for state in self._node_states:
+            if state.name == 'DONE':
+                continue
+            node_filters['state'] = state.value
```

Comment:
The state names are fixed for a given API version, so this should be initialised in the constructor rather than every time the method is called.  Something like this I guess:
```python
self._pending_states = [state for state in self._db.node_states if state.name != 'DONE']
```
Also I think there's an Enum now we can use instead of the plain `DONE` string.

Reply:
We could even have the node filter created in the constructor as that's also constant.

Reply:
Rebased and updated the list to directly have `state.value`.